Accept model type parameter in export_llama

jackzhxng opened the pull request:
### Summary
Specify model to export in the CLI.

### Test plan
Exported the stories 110M model.
```
python -m examples.models.llama.export_llama -c stories110M/stories110M.pt -p stories110M/params.json -X -kv
```

PR chain:
- [Add kwarg example inputs to eager model base](https://github.com/pytorch/executorch/pull/5765)
- [Llama2 model cleanup](https://github.com/pytorch/executorch/pull/5859)
- **YOU ARE HERE ~>** [Accept model type parameter in export_llama](https://github.com/pytorch/executorch/pull/5910)
- [Export TorchTune llama3_2_vision in ET](https://github.com/pytorch/executorch/pull/5911)
- [Runner changes for TorchTune Llama3.2 vision text decoder](https://github.com/pytorch/executorch/pull/6610)

helunwencser commented on examples/models/llama/export_llama_lib.py:
```diff
@@ -128,6 +132,12 @@ def build_args_parser() -> argparse.ArgumentParser:
+        default="llama3",
+        choices=EXECUTORCH_DEFINED_MODELS + TORCHTUNE_DEFINED_MODELS,
+        help="The Lllama model to export. llama2, llama3, llama3_1, llama3_2 share the same architecture, so they are technically interchangeable given you provide the checkpoint file for the desired version.",
+    )
```
I'm wondering what is the motivation for adding this command line option? All llama models share the same model architecture and code. The `modelName` in the codebase does not really do any thing. Everything should continue to work if we hardcode a name for it or remove it. On the other hand, if we add one more command line option, it adds more work for people who need to use `export_llama`, `eval_llama` commands by making them have more options to add.

I feel we should change `class Llama2Model` to `class LlamaModel` instead.

helunwencser commented on examples/models/llama/export_llama_lib.py:
```diff
@@ -114,11 +118,11 @@ def build_model(
         output_dir_path = "."
 
-    argString = f"--checkpoint par:{modelname}_ckpt.pt --params par:{modelname}_params.json {extra_opts} --output-dir {output_dir_path}"
+    argString = f"--model {modelname} --checkpoint par:{modelname}_ckpt.pt --params par:{modelname}_params.json {extra_opts} --output-dir {output_dir_path}"
```
it will not work here because currently `modelname` is set to `model`, for example, https://github.com/pytorch/executorch/blob/main/examples/models/llama/llama_test.py#L15-L23, while you expect modelname to be one of the `["llama2", "llama3", "llama3_1", "llama3_2"]`

helunwencser commented on examples/models/llama/export_llama_lib.py:
```diff
@@ -785,15 +795,29 @@ def _load_llama_model(
     Returns:
         An instance of LLMEdgeManager which contains the eager mode model.
     """
+
```
unexpected change

helunwencser commented on examples/models/llama/export_llama_lib.py:
```diff
@@ -785,15 +795,29 @@ def _load_llama_model(
+    if modelname in EXECUTORCH_DEFINED_MODELS:
+        # Set to llama2 because all models in EXECUTORCH_DEFINED_MODELS share the same archteciture as
+        # defined in example/models/llama2.
+        modelname = "llama"
```
why not set modelname to `modelname`?

helunwencser commented on examples/models/llama/export_llama_lib.py:
```diff
@@ -785,15 +795,29 @@ def _load_llama_model(
+            "Torchtune Llama models are not yet supported in ExecuTorch export."
+        )
+    else:
+        raise ValueError(f"{modelname} is not a valid Llama model.")
```
I believe this will break some existing tests as I mentioned above because some tests use `model` as the name.

helunwencser commented on .ci/scripts/test_eval_llama_mmlu.sh:
```diff
@@ -35,6 +35,7 @@ run_and_verify() {
         exit 1
     fi
     $PYTHON_EXECUTABLE -m examples.models.llama.eval_llama \
+	--model llama2 \
```
this seems weird because we are actually running the stories model here but we use llama2 as the name.

Commit:
Accept model type parameter in export_llama (#6507)

Summary:
Specify model to export in the CLI.


Test Plan:
Exported the stories 110M model.
```
python -m examples.models.llama.export_llama -c stories110M/stories110M.pt -p stories110M/params.json -X -kv
```

PR chain:
- [Add kwarg example inputs to eager model base](https://github.com/pytorch/executorch/pull/5765)
- [Llama2 model cleanup](https://github.com/pytorch/executorch/pull/5859)
- **YOU ARE HERE ~>** [Accept model type parameter in export_llama](https://github.com/pytorch/executorch/pull/5910)
- [Export TorchTune llama3_2_vision in ET](https://github.com/pytorch/executorch/pull/5911)
- [Runner changes for TorchTune Llama3.2 vision text decoder](https://github.com/pytorch/executorch/pull/6610)
- [Add et version of TorchTune MHA for swapping with custom op](https://github.com/pytorch/executorch/pull/5912)

Reviewed By: helunwencser

Differential Revision: D65612837

Pulled By: dvorjackz
